src/server/internalServer_catalog_v2.cpp

```diff
@@ -146,4 +148,18 @@ std::unique_ptr<Response> InternalServer::handle_catalog_v2_languages(const Requ
   );
 }
 
+std::unique_ptr<Response> InternalServer::handle_catalog_v2_illustration(const RequestContext& request)
+{
+  try {
+    const auto bookName  = request.get_url_part(3);
+    const auto bookId = mp_nameMapper->getIdForName(bookName);
+    auto book = mp_library->getBookById(bookId);
```

Comment:
Since library reloading is now supported we should use library in a thread-safe way. `Library::getBookById()` is not a thread-safe method. Use `Library::getBookByIdThreadSafe()` instead.